server/store.py

```diff
@@ -0,0 +1,337 @@
+            tasks.append(asyncio.create_task(
+                self.index_daily_recording(sem, recording)))
+        # Wait for tasks to complete
+        await asyncio.gather(*tasks)
+
+    async def index_daily_recording(self, sem: asyncio.locks.BoundedSemaphore,
+                                    recording: Recording):
+        """Indexes given Daily recording"""
+        async with sem:
+            loop = asyncio.get_event_loop()
+            await loop.run_in_executor(None, self.transcribe_and_index_recording, recording)
+
+    async def index_uploaded_file(self, sem: asyncio.locks.BoundedSemaphore, file_path: str):
+        """Indexes given uploaded file"""
+        async with sem:
+            loop = asyncio.get_event_loop()
+            await loop.run_in_executor(None, self.transcribe_and_index_file, file_path)
+
+    def transcribe_and_index_file(self, video_path):
+        """Transcribes and indexes locally-saved video recording."""
+        file_name = pathlib.Path(video_path).stem
+
+        # Set up transcript file names and paths
+        transcript_file_name = f"{file_name}.txt"
+        transcripts_dir = get_transcripts_dir_path()
+        transcript_file_path = os.path.join(
+            transcripts_dir, transcript_file_name)
+
+        # Don't re-transcribe if a transcript for this recording already exists
+        if os.path.exists(transcript_file_path):
+            os.remove(video_path)
+            return
+
+        # If audio for this video does not already exist, extract it.
+        audio_path = get_audio_path(video_path)
+        if not os.path.exists(audio_path):
+            audio_path = extract_audio(video_path)
+
+        # Video no longer needed, remove it.
+        os.remove(video_path)
+
+        # Transcribe the generated audio file.
+        transcript = self.transcriber.transcribe(None, audio_path)
+
+        # Save the new transcript and _maybe_ index it, if relevant.
+        self.save_and_index_transcript(transcript_file_path, transcript)
+
+        # No need to take up disk space with the audio file
+        # once we have the transcript saved
+        if audio_path:
+            os.remove(audio_path)
+
+    def transcribe_and_index_recording(self, recording: Recording):
+        """Transcribes and indexes a Daily cloud recording"""
+
+        print("Transcribing and indexing recording:", recording)
+        # A safety rail to make sure we only include relevant room name
+        # In reality, we specify the room name when querying Daily's REST API
+        if self.daily_room_name and self.daily_room_name != recording.room_name:
+            return
+
+        file_name = f"{recording.timestamp}_{recording.room_name}_{recording.id}"
+
+        # Set up transcript file names and paths
+        transcript_file_name = f"{file_name}.txt"
+        transcripts_dir = get_transcripts_dir_path()
+        transcript_file_path = os.path.join(
+            transcripts_dir, transcript_file_name)
```

Comment:
may be a function to get the transcript file name since we use it in `transcribe_and_index_file` as well. just in case we change in one place but not the other.

Reply:
```
def get_transcript_file_path(file_name):
        transcript_file_name = f"{file_name}.txt"
        transcripts_dir = get_transcripts_dir_path()
        return os.path.join(transcripts_dir, transcript_file_name)
```

Reply:
Good idea